pmd-javascript/src/main/java/net/sourceforge/pmd/lang/ecmascript/ast/ASTUnaryExpression.java

```diff
@@ -32,10 +28,10 @@ public Object jjtAccept(EcmascriptParserVisitor visitor, Object data) {
     }
 
     public boolean isPrefix() {
-        return node.isPrefix();
+        return !isPostfix();
     }
 
     public boolean isPostfix() {
-        return node.isPostfix();
+        return node.getOperator() == Token.INC || node.getOperator() == Token.DEC;
```

Comment:
from what I understand unaries with these tokens are now parsed as `ASTUpdateExpression` nodes, so shouldn't this return false always?

Reply:
Probably - I didn't investigate further. I only noticed that some unaryexpressions now appeared as updateexpressions.